contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,850 @@
+guarantees.
+
+1. The controller will 
+[reconstruct the revision history](#history-reconstruction) of the StatefulSet.
+1. The controller will process any [template updates](#template-updates) to 
+ensure that the StatefulSet's revision history is consistent with the user 
+declared desired state.
+1. The controller will select all Pods in the StatefulSet, filter any Pods not 
+owned by the StatefulSet, and sort the remaining Pods in ordinal order.
+1. If any Pods with ordinals in the sequence `[0,.Spec.Replicas)` have not been 
+created, for the Pod corresponding to the lowest such ordinal, the controller 
+will [select the PodTemplate](#podtemplate-selection) from the StatefulSet's 
+revision history corresponding to the Pod's ordinal and create the Pod.
+1. If all Pods in the sequence `[0,.Spec.Replicas)` have been created, but if any 
+have a Status other than Running or do not have a Ready Condition, the 
+StatefulSet controller will wait for these Pods to either become Running and 
+Ready, or to be completely deleted. 
+1. If all Pods in the sequence `[0,.Spec.Replicas)` have a Status of Running and 
+a Condition indicating Ready, and if `.Spec.Replicas` is less than 
+`.Status.Replicas`, the controller will delete the Pod corresponding to the 
+largest ordinal. This implies that scaling takes precedence over Pod updates.
+1. If all Pods in the range `[0,.Spec.Replicas)` have a Status of Running and 
+a Ready Condition, if `.Spec.Replicas` is equal to `.Status.Replicas`, and if 
+there are Pods that do not match the 
+[declared desired PodTemplate](#podtemplate-selection), the Pod corresponding to 
+the largest ordinal will be deleted.
+1. If the StatefulSet controller has achieved the 
+[declared target state](#target-state), and if that state has a 
+`.Spec.ParitionOrdinal` equal to `0`, the StatefulSet controller will 
+[complete any in progress updates](#update-completion).
+1. The controller will [report its status](#status-reporting).
+1. The controller will perform any necessary
+[maintenance of its revision history](#history-maintenance).
+
+### StatefulSet Revision History
+The StatefulSet controller will use labeled, versioned PodTemplates to keep a 
+history of updates preformed on a StatefulSet. The number of stored PodTemplates 
+is considered to be the depth of the StatefulSet's revision history. The 
+maximum revision history depth for a StatefulSet must be at least two, but it 
+may be greater.
+
+#### PodTemplate Creation
+When the StatefulSet controller creates a PodTemplate for a StatefulSet, it will 
+do the following.
+
+1. The controller will set the PodTemplate's `.PodTemplateSpec` field to the 
+StatefulSet's `.Spec.Template` field.
+1. The controller will create a ControllerRef object in the PodTemplate's 
+`.OwnerReferences` list to mediate selector overlapping.
+1. The controller will label the PodTemplate with a 
```

Comment:
The point above still applies. The only way this should happen is if the user explicitly mutates the label applied to the PodTemplate, or, if StatefulSet.Spec.Selector becomes mutable, by mutating the selector. Are we worried about orphan's occurring due to unintentional user error? If so, lets not make the selector mutable.
Also, I was not planning on having controllers adopt PodTemplates in the same way they adopt orphan Pods. A controller taking ownership of an orphaned Pod makes sense, a controller taking ownership of another controllers revision history, imo, does not. I don't think we need to solve for the same constraints as we did for Pods.